deeplake/core/vectorstore/vector_search/vector_search.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+
+from typing import Any, Callable, Dict, List, Union
+
+import deeplake
+from deeplake.core.dataset import Dataset as DeepLakeDataset
+from deeplake.core import vectorstore
+
+
+EXEC_OPTION_TO_SEARCH_TYPE: Dict[str, Callable] = {
+    "compute_engine": vectorstore.indra_vector_search,
+    "python": vectorstore.python_vector_search,
+    "db_engine": vectorstore.remote_engine_vector_search,
+}
+
+
+def search(
+    query_embedding: Union[List[float], np.ndarray],
+    embedding: Union[List[float], np.ndarray],
+    k: int,
+    distance_metric: str,
+    exec_option: str,
+    deeplake_dataset: DeepLakeDataset,
+    embedding_tensor: str = "embedding",
+):
+    """Searching function
+    Args:
+        query_embedding (Union[List[float], np.ndarray]) - embedding representation of the query
+        embedding (Union[List[float], np.ndarray) - full embeddings representation of the dataset, used only in python implementation.
+        k (int) - number of samples to return after searching
+        distance_metric (str, optional): Type of distance metric to use for sorting the data. Avaliable options are: "L1", "L2", "COS", "MAX".
+        exec_option (str, optional): Type of query execution. It could be either "python", "compute_engine" or "db_engine". Defaults to "python".
+            - `python` - runs on the client and can be used for any data stored anywhere. WARNING: using this option with big datasets is discouraged, because it can lead to some memory issues.
+            - `compute_engine` - runs on the client and can be used for any data stored in or connected to Deep Lake.
```

Comment:
single backtick (`) will make the text italic, we should use double backtick (``) for this kind of text.